### PR TITLE
Add filter tests to Jetscape tests

### PR DESCRIPTION
**Pull request by Hendrik1704**
This adds the missing tests to the Jetscape tests.

@nilssass I chose to do it with the pytest fixtures for the particle lists, this is a bit cleaner. You can copy this for the Oscar tests if you want.

### CODE REVIEW

**Comment by Hendrik1704** [comment]
There was some output from tests written to the main sparkx directory. I redirected this output to `tmp_path`.

**Comment by Hendrik1704** [comment]
I think, that the Jetscape tests are complete now. I found a bug in the code, which is also fixed here. When applying the kwargs filter in Jetscape, these were not applied to the last event in the file.

File: src/sparkx/Jetscape.py [hunk 1]
```diff
@@ -389,7 +392,7 @@ def set_particle_list(self, kwargs):
                     else:
                         if 'filters' in self.optional_arguments_.keys():
                             data = self.__apply_kwargs_filters([data],kwargs['filters'])[0]
-                            self.num_output_per_event_[len(particle_list)]=(len(particle_list),len(data))
+                            self.num_output_per_event_[len(particle_list)]=(len(particle_list)+1,len(data))
```
Comment (NGoetz) [inline src/sparkx/Jetscape.py:395]: Are we sure this bug is not present in Oscar as well?

**Review by NGoetz — APPROVED** [review]
Looks good, we should just check that this bug was JETSCAPE specific.

**Comment by Hendrik1704** [comment]
I think this is purely in Jetscape. The function in the Oscar class is completely different due to the different file structure in the inputs.

**Pull request merged**